### PR TITLE
feat(pagination): Add component tokens

### PR DESCRIPTION
**Pull request by macandcheese**
**Related Issue:** #7180 

## Summary

`--calcite-pagination-background-color-active:` Specifies the background color of the pagination items while active.
`--calcite-pagination-background-color-hover:` Specifies the background color of the pagination items while hovered.
`--calcite-pagination-background-color`: Specifies the background color of the pagination items.
`--calcite-pagination-border-color-hover:` Specifies the border color the pagination items while hovered.
`--calcite-pagination-border-color-selected:` Specifies the border color of the pagination items when selected.
`--calcite-pagination-text-color-hover:` Specifies the text color of the pagination items items while hovered.
`--calcite-pagination-text-color-selected:` Specifies the text color of the pagination items when selected.
`--calcite-pagination-text-color: Specifies` the text color of the pagination items.
`--calcite-pagination-arrow-background-color-active`: Specifies the background color of the arrow items while active.
`--calcite-pagination-arrow-background-color-hover`: Specifies the background color of the arrow items while hovered.
`--calcite-pagination-arrow-background-color:` Specifies the background color of the arrow items.
`--calcite-pagination-arrow-icon-color-active`: Specifies the icon color of the arrow items while active.
`--calcite-pagination-arrow-icon-color-hover`: Specifies the icon color of the arrow items while hovered.
`--calcite-pagination-arrow-icon-color:` Specifies the icon color of the arrow items.

Adds local demo and Chromatic screenshot test.

### CODE REVIEW

File: packages/calcite-components/src/components/pagination/pagination.scss [hunk 1]
```diff
@@ -1,42 +1,86 @@
...
   & .page,
   & .ellipsis {
-    @apply text-n1h h-8 px-2;
-    min-inline-size: theme("width.8");
+    @apply text-n1h;
   }
 }
 
 :host([scale="l"]) {
+  --calcite-internal-pagination-item-size: 44px;
```
Comment (macandcheese) [inline packages/calcite-components/src/components/pagination/pagination.scss:72]: @alisonailea Seems like I can't get a variable with this value to work - directly importing the file that has the `size-44` reference doesn't seem to work, are any examples of these variables being used in component css files?

File: packages/calcite-components/src/components/pagination/pagination.scss [hunk 2]
```diff
@@ -1,42 +1,88 @@
+/**
+ * CSS Custom Properties
+ *
+ * These properties can be overridden using the component's tag as selector.
+ *
+
+ * @prop --calcite-pagination-background-color-active: Specifies the background color of the pagination items while active.
+ * @prop --calcite-pagination-background-color-hover: Specifies the background color of the pagination items while hovered.
```
Comment (alisonailea) [inline packages/calcite-components/src/components/pagination/pagination.scss:8]: remove the stateful tokens here.

File: packages/calcite-components/src/components/pagination/pagination.scss (hunk 1 above)
Comment (alisonailea) [reply packages/calcite-components/src/components/pagination/pagination.scss:74]: you can do an `@import '~@esri/calcite-design-tokens/dist/scss/core.scss'`

File: packages/calcite-components/src/components/pagination/pagination.scss [hunk 3]
```diff
@@ -1,42 +1,88 @@
...
+ * @prop --calcite-pagination-border-color-selected: Specifies the border color of the pagination items when selected.
+ * @prop --calcite-pagination-text-color-hover: Specifies the text color of the pagination items items while hovered.
+ * @prop --calcite-pagination-text-color-selected: Specifies the text color of the pagination items when selected.
+ * @prop --calcite-pagination-text-color: Specifies the text color of the pagination items.
+ * @prop --calcite-pagination-arrow-background-color-active: Specifies the background color of the arrow items while active.
+ * @prop --calcite-pagination-arrow-background-color-hover: Specifies the background color of the arrow items while hovered.
+ * @prop --calcite-pagination-arrow-background-color: Specifies the background color of the arrow items.
+ * @prop --calcite-pagination-arrow-icon-color-active: Specifies the icon color of the arrow items while active.
+ * @prop --calcite-pagination-arrow-icon-color-hover: Specifies the icon color of the arrow items while hovered.
+ * @prop --calcite-pagination-arrow-icon-color: Specifies the icon color of the arrow items.
```
Comment (alisonailea) [inline packages/calcite-components/src/components/pagination/pagination.scss:20]: are these state tokens necessary?

**Review by alisonailea — CHANGES_REQUESTED** [review]

File: packages/calcite-components/src/components/pagination/pagination.scss (hunk 2 above)
Comment (macandcheese) [reply packages/calcite-components/src/components/pagination/pagination.scss:8]: These are rendered within the component - without this the user can't style on hover.

File: packages/calcite-components/src/components/pagination/pagination.scss (hunk 3 above)
Comment (macandcheese) [reply packages/calcite-components/src/components/pagination/pagination.scss:20]: Same as above - a user can't override these styles without these props as these are rendered within the component.

File: packages/calcite-components/src/components/pagination/pagination.scss (hunk 1 above)
Comment (macandcheese) [reply packages/calcite-components/src/components/pagination/pagination.scss]: Updated and used the variable - can I get another look? Thanks

**Pull request merged**